src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs

```diff
@@ -64,6 +64,34 @@ public class PublicationService : IPublicationService
             _themeCacheService = themeCacheService;
         }
 
+        public async Task<Either<ActionResult, List<PublicationViewModel>>> GetPublicationsByTopic(
+            bool permissions,
+            Guid? topicId)
+        {
+            var userId = _userService.GetUserId();
+
+            return await _userService
+                .CheckCanAccessSystem()
+                .OnSuccess(_ => _userService.CheckCanViewAllPublications()
+                    .OnSuccess(() =>
+                    {
+                        if (!topicId.HasValue)
+                        {
+                            throw new ArgumentException(
+                                "Users with permissions to view all releases shouldn't request all publications");
```

Comment:
> Should this error be view all publications instead?
 :+1:

Reply:
> Two methods GetAllPublications(bool permissions = false) and GetPublicationsByTopic(Guid topicId, bool permissions = false) might help make this clearer and make the testing clearer too.

Discussed this with Ben and Nick on Slack, and we agreed to leave things as they are. We considered this split, but even by doing we would still need the `OrElse` code. This is because the mechanism by which a non-BAU users's publications are fetched and a BAU user's publications are fetched are different. For BAU, we just fetch all the publications, while with non-BAU we check existing user release and publication roles. For this splitting to work, we'd really want the same mechanism/code path for getting publications for all users.